src/actions/editorActions.js

```diff
@@ -41,41 +42,31 @@ export function recover() {
 */
 export function fetchScene(id, uid = "anon") {
     return (dispatch) => {  // Return a func that dispatches events after async
-        scenes.doc(id).get().then((scene) => {
-            let data = scene.data();
-            if (data && data.pw) {
-                let pw = prompt("Please enter the PW");
-                if (pw !== data.pw) { return; }
-            }
-            if (data && data.code) { // If it worked
-                // render the editor
-                dispatch(render(data.code, uid || "anon"));
-                dispatch(updateSavedText(data.code));
-                // Use default for eventual consistency in db
-                let settings = DEF_SETTINGS;
-
-                // if the incoming scene has a settings, merge default with incoming
-                if (data.settings) {
-                    settings = { ...settings, ...data.settings };
+        fetch(`${sceneRef}/id/${id}`).then((response) =>{
+            if(response.status !== 200){
+                if(response.status === 404){
+                    window.location.href = window.origin + "/error-404";
                 }
+            }
```

Comment:
Is there a reason why we're checking for not 200 and then for 404 instead of checking 404 directly? Or should we be checking for more than just 404?

Reply:
Forgot to check for other errors, fixing in upcoming commit